File: NightshadeBerries/NightshadeBerries.cpp
#include "NightshadeBerries.h"
#include "../Settings.h"

NightshadeBerries::NightshadeBerries(World *world, int x, int y)
    : Plant(NIGHTSHADE_BERRIES_STRENGTH, NIGHTSHADE_BERRIES_SYMBOL, NIGHTSHADE_BERRIES_COLOR, world, x, y) {}

void NightshadeBerries::action() {
  Plant::action();
}

void NightshadeBerries::collision(Organism* organism) {
  organism->setAlive(false);
  this->getWorld()->addLog("%s POS:(%d, %d) killed %s POS:(%d, %d)", this->getType().c_str(), this->getPosition().getX(), this->getPosition().getY(), organism->getType().c_str(), organism->getPosition().getX(), organism->getPosition().getY());
}

std::string NightshadeBerries::getType() const { return "NightshadeBerries"; }

Organism* NightshadeBerries::createChild(int x, int y, int parent1Id, int parent2Id) {
  return new NightshadeBerries(this->getWorld(), x, y);
}

File: src/NightshadeBerries/NightshadeBerries.java
package NightshadeBerries;

import Field.Field;
import Guarana.Guarana;
import Organism.Organism;
import Organism.OrganismData;
import Plant.Plant;
import Settings.Settings;
import World.World;

public class NightshadeBerries extends Plant {

    public NightshadeBerries(World world, Field field) {
        super(new OrganismData(world.getNextOrganismId(), Settings.NIGHTSHADE_BERRIES_INITIATIVE, Settings.NIGHTSHADE_BERRIES_STRENGTH, Settings.NIGHTSHADE_BERRIES_IMAGE, -1, -1), world, field);
    }

    @Override
    public void action() {
        super.action();
    }

    @Override
    public void collision(Organism attacker) {
        if (attacker.isAnimal()) {
            this.world.addLog(this.getType() + "(" + this.getCurrentField().getNumber() + ") killed " + attacker.getType());
            attacker.getData().setAlive(false);
        }
    }

    @Override
    public String getType() {
        return "NightshadeBerries";
    }

    @Override
    public void reproduce(Field field) {
        Organism nightshadeBerries = new NightshadeBerries(this.world, field);
        this.world.addOrganism(nightshadeBerries);
        this.world.addLog(this.getType() + " reproduced (" + field.getNumber() + ")");
    }

}
